### ISSUE TITLE
Gimp crashes during a splash screen

### ISSUE THREAD

**Issue by KristijanZic** [aferrero2707/gimp-appimage#19]
When I run the GIMP_AppImage-git-2.10.7-withplugins-20180908-x86_64.AppImage or GIMP_AppImage-git-2.10.7-20180908-x86_64.AppImage i get this error:

`$ ./GIMP_AppImage-git-2.10.7-withplugins-20180908-x86_64.AppImage 
DIR: /tmp/.mount_GIMP_ACCsa6n
AILIBDIR=/tmp/tmp.wKPETi85Fj
deleting /tmp/tmp.wKPETi85Fj/libxcb-dri3.so*
System stdc++ library: "/usr/lib/x86_64-linux-gnu/libstdc++.so.6"
System stdc++ library version: "3.4.25"
Bundled stdc++ library version: "3.4.22"
Newest stdc++ library version: "3.4.25"
Using system stdc++ library
atomiclib: /usr/lib/x86_64-linux-gnu/libatomic.so.1
Checking versions of library "libfontconfig"
  system library: "/usr/lib/x86_64-linux-gnu/libfontconfig.so.1.10.1" (libfontconfig.so.1.10.1)
  system library version: 1.10.1
  ailib: "/tmp/.mount_GIMP_ACCsa6n/usr/lib/libfontconfig.so.1.11.1"
  bundled library: "/tmp/.mount_GIMP_ACCsa6n/usr/lib/libfontconfig.so.1.11.1" (libfontconfig.so.1.11.1)
  bundled library version: 1.11.1
  newest library version: "1.11.1"
Checking versions of library "libfreetype"
  system library: "/usr/lib/x86_64-linux-gnu/libfreetype.so.6.15.0" (libfreetype.so.6.15.0)
  system library version: 6.15.0
  ailib: "/tmp/.mount_GIMP_ACCsa6n/usr/lib/libfreetype.so.6.16.1"
  bundled library: "/tmp/.mount_GIMP_ACCsa6n/usr/lib/libfreetype.so.6.16.1" (libfreetype.so.6.16.1)
  bundled library version: 6.16.1
  newest library version: "6.16.1"
GDK_PIXBUF_MODULE_FILE: /tmp/tmp.wKPETi85Fj/gdk-pixbuf-2.0/loaders.cache
GTK_PATH=/tmp/.mount_GIMP_ACCsa6n/usr/lib/gtk-2.0
GTK_IM_MODULE_FILE=/tmp/.mount_GIMP_ACCsa6n/usr/lib/gtk-2.0:/tmp/.mount_GIMP_ACCsa6n/usr/lib/gtk-2.0
PANGO_LIBDIR=/tmp/.mount_GIMP_ACCsa6n/usr/lib
BABL_PATH /tmp/.mount_GIMP_ACCsa6n/usr/lib/babl-0.1
GEGL_PATH /tmp/.mount_GIMP_ACCsa6n/usr/lib/gegl-0.4
python=/tmp/.mount_GIMP_ACCsa6n/usr/bin/python2
python2=/tmp/.mount_GIMP_ACCsa6n/usr/bin/python2
/tmp/.mount_GIMP_ACCsa6n/usr/bin/python=/tmp/.mount_GIMP_ACCsa6n/usr/bin/python2
:Python:E::py::python2:
GIMP2_PLUGINDIR: /tmp/tmp.wKPETi85Fj/gimp/2.0
Input parameters: ""

Input File: 
/bin/bash: /tmp/tmp.wKPETi85Fj/libtinfo.so.5: no version information available (required by /bin/bash)
	linux-vdso.so.1 (0x00007ffcf3f44000)
	libgimpwidgets-2.0.so.0 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libgimpwidgets-2.0.so.0 (0x00007febcb010000)
	libgtk-x11-2.0.so.0 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libgtk-x11-2.0.so.0 (0x00007febca98e000)
	libgdk-x11-2.0.so.0 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libgdk-x11-2.0.so.0 (0x00007febca6cd000)
	libatk-1.0.so.0 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libatk-1.0.so.0 (0x00007febca4a7000)
	libgimpconfig-2.0.so.0 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libgimpconfig-2.0.so.0 (0x00007febca291000)
	libgimpmath-2.0.so.0 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libgimpmath-2.0.so.0 (0x00007febca08b000)
	libgimpthumb-2.0.so.0 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libgimpthumb-2.0.so.0 (0x00007febc9e81000)
	libgimpcolor-2.0.so.0 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libgimpcolor-2.0.so.0 (0x00007febc9c69000)
	libgimpmodule-2.0.so.0 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libgimpmodule-2.0.so.0 (0x00007febc9a63000)
	libgimpbase-2.0.so.0 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libgimpbase-2.0.so.0 (0x00007febc9834000)
	libgdk_pixbuf-2.0.so.0 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libgdk_pixbuf-2.0.so.0 (0x00007febc960c000)
	libpangocairo-1.0.so.0 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libpangocairo-1.0.so.0 (0x00007febc93ff000)
	libpangoft2-1.0.so.0 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libpangoft2-1.0.so.0 (0x00007febc91e9000)
	libpango-1.0.so.0 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libpango-1.0.so.0 (0x00007febc8f9e000)
	libcairo.so.2 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libcairo.so.2 (0x00007febc8c85000)
	libpixman-1.so.0 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libpixman-1.so.0 (0x00007febc89dc000)
	libfontconfig.so.1 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libfontconfig.so.1 (0x00007febc879a000)
	libexpat.so.1 => /lib/x86_64-linux-gnu/libexpat.so.1 (0x00007febc8568000)
	libuuid.so.1 => /lib/x86_64-linux-gnu/libuuid.so.1 (0x00007febc8361000)
	libfreetype.so.6 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libfreetype.so.6 (0x00007febc80ba000)
	libbz2.so.1 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libbz2.so.1 (0x00007febc7eaa000)
	libharfbuzz.so.0 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libharfbuzz.so.0 (0x00007febc7c47000)
	libpng16.so.16 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libpng16.so.16 (0x00007febc7a14000)
	libxcb-shm.so.0 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libxcb-shm.so.0 (0x00007febc7810000)
	libxcb.so.1 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libxcb.so.1 (0x00007febc75e8000)
	libxcb-render.so.0 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libxcb-render.so.0 (0x00007febc73da000)
	libXrender.so.1 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libXrender.so.1 (0x00007febc71cf000)
	libX11.so.6 => /usr/lib/x86_64-linux-gnu/libX11.so.6 (0x00007febc6e97000)
	libXext.so.6 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libXext.so.6 (0x00007febc6c85000)
	librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007febc6a7d000)
	libgegl-npd-0.4.so => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libgegl-npd-0.4.so (0x00007febc6875000)
	libgegl-0.4.so.0 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libgegl-0.4.so.0 (0x00007febc65ad000)
	libgthread-2.0.so.0 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libgthread-2.0.so.0 (0x00007febc63ab000)
	libgmodule-2.0.so.0 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libgmodule-2.0.so.0 (0x00007febc61a7000)
	libjson-glib-1.0.so.0 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libjson-glib-1.0.so.0 (0x00007febc5f7f000)
	libbabl-0.1.so.0 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libbabl-0.1.so.0 (0x00007febc5ca3000)
	libgio-2.0.so.0 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libgio-2.0.so.0 (0x00007febc5908000)
	liblcms2.so.2 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/liblcms2.so.2 (0x00007febc56af000)
	libgexiv2.so.2 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libgexiv2.so.2 (0x00007febc548d000)
	libexiv2.so.26 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libexiv2.so.26 (0x00007febc4e9d000)
	libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007febc4b0f000)
	libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007febc48f2000)
	libmypaint-1.3.so.0 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libmypaint-1.3.so.0 (0x00007febc46e0000)
	libjson-c.so.2 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libjson-c.so.2 (0x00007febc44d5000)
	libgobject-2.0.so.0 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libgobject-2.0.so.0 (0x00007febc4284000)
	libglib-2.0.so.0 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libglib-2.0.so.0 (0x00007febc3f70000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007febc3bd2000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007febc39ce000)
	libjpeg.so.62 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libjpeg.so.62 (0x00007febc3779000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007febc355a000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007febc3169000)
	libXfixes.so.3 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libXfixes.so.3 (0x00007febc2f63000)
	libXinerama.so.1 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libXinerama.so.1 (0x00007febc2d60000)
	libXi.so.6 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libXi.so.6 (0x00007febc2b50000)
	libXrandr.so.2 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libXrandr.so.2 (0x00007febc2945000)
	libXcursor.so.1 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libXcursor.so.1 (0x00007febc273a000)
	libXcomposite.so.1 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libXcomposite.so.1 (0x00007febc2537000)
	libXdamage.so.1 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libXdamage.so.1 (0x00007febc2334000)
	libpng15.so.15 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libpng15.so.15 (0x00007febc2109000)
	libthai.so.0 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libthai.so.0 (0x00007febc1efd000)
	/lib64/ld-linux-x86-64.so.2 (0x00007febcb2a0000)
	libgraphite2.so.3 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libgraphite2.so.3 (0x00007febc1ccf000)
	libXau.so.6 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libXau.so.6 (0x00007febc1acb000)
	libpcre.so.1 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libpcre.so.1 (0x00007febc1869000)
	libffi.so.6 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libffi.so.6 (0x00007febc1661000)
	libselinux.so.1 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libselinux.so.1 (0x00007febc143a000)
	libresolv.so.2 => /lib/x86_64-linux-gnu/libresolv.so.2 (0x00007febc121f000)
	libmount.so.1 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libmount.so.1 (0x00007febc0fdd000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007febc0dc5000)
	libblkid.so.1 => /tmp/.mount_GIMP_ACCsa6n/usr/bin/../lib/libblkid.so.1 (0x00007febc0b86000)

/tmp/.mount_GIMP_ACCsa6n/usr/bin/gimp.bin --pdb-compat-mode=on ""
/tmp/.mount_GIMP_ACCsa6n/usr/bin/python
Gtk-Message: Failed to load module "gail"
Gtk-Message: Failed to load module "atk-bridge"

(gimp.bin:9002): Gtk-WARNING **: Unable to locate theme engine in module_path: "adwaita",

(gimp.bin:9002): Gtk-WARNING **: Unable to locate theme engine in module_path: "adwaita",
Gtk-Message: Failed to load module "unity-gtk-module"
gimp_device_info_set_device: trying to set GdkDevice 'Mad Catz Mad Catz R.A.T. Pro X' on GimpDeviceInfo which already has a device
/tmp/.mount_GIMP_ACCsa6n/usr/bin/gimp.bin: fatal error: Segmentation fault
27	../sysdeps/unix/sysv/linux/read.c: No such file or directory.
`

**Comment by aferrero2707** [comment]
Which distribution type/version are you using?

**Comment by aferrero2707** [comment]
Duplicate of #20

**Comment by KristijanZic** [comment]
`$ neofetch
            .-/+oossssoo+/-.               aresminos@desktop 
        `:+ssssssssssssssssss+:`           ----------------- 
      -+ssssssssssssssssssyyssss+-         OS: Ubuntu 18.04.1 LTS x86_64 
    .ossssssssssssssssssdMMMNysssso.       Kernel: 4.18.8-041808-generic 
   /ssssssssssshdmmNNmmyNMMMMhssssss/      Uptime: 2 hours, 14 mins 
  +ssssssssshmydMMMMMMMNddddyssssssss+     Packages: 2571 
 /sssssssshNMMMyhhyyyyhmNMMMNhssssssss/    Shell: bash 4.4.19 
.ssssssssdMMMNhsssssssssshNMMMdssssssss.   Resolution: 1680x1050 
+sssshhhyNMMNyssssssssssssyNMMMysssssss+   DE: GNOME 3.28.3 
ossyNMMMNyMMhsssssssssssssshmmmhssssssso   WM: GNOME Shell 
ossyNMMMNyMMhsssssssssssssshmmmhssssssso   WM Theme: Adwaita 
+sssshhhyNMMNyssssssssssssyNMMMysssssss+   Theme: Communitheme [GTK2/3] 
.ssssssssdMMMNhsssssssssshNMMMdssssssss.   Icons: Suru [GTK2/3] 
 /sssssssshNMMMyhhyyyyhdNMMMNhssssssss/    Terminal: gnome-terminal 
  +sssssssssdmydMMMMMMMMddddyssssssss+     CPU: Intel i5-2400 (4) @ 3.400GHz 
   /ssssssssssshdmNNNNmyNMMMMhssssss/      GPU: AMD Radeon RX Vega 
    .ossssssssssssssssssdMMMNysssso.       Memory: 3051MiB / 3927MiB 
      -+sssssssssssssssssyyyssss+- 
        `:+ssssssssssssssssss+:`                                   
`

**Comment by KristijanZic** [comment]
aresminos@desktop 
----------------- 
OS: Ubuntu 18.04.1 LTS x86_64 
Kernel: 4.18.8-041808-generic 
Uptime: 2 hours, 14 mins 
Packages: 2571 
Shell: bash 4.4.19 
Resolution: 1680x1050 
DE: GNOME 3.28.3 
WM: GNOME Shell 
WM Theme: Adwaita 
Theme: Communitheme [GTK2/3] 
Icons: Suru [GTK2/3] 
Terminal: gnome-terminal 
CPU: Intel i5-2400 (4) @ 3.400GHz
GPU: AMD Radeon RX Vega 
Memory: 3051MiB / 3927MiB

**Comment by aferrero2707** [comment]
@KristijanZic 
Please see the discussion in issue #20: the problem seems to be due to some USB devices being plugged when GIMP is started, and is probably not specific to the AppImage itself.

Could you try to unplug your 'Mad Catz Mad Catz R.A.T. Pro X' before starting GIMP, and plug it back when it is launched?

Thanks!

**Comment by KristijanZic** [comment]
@aferrero2707 
After unplugging 'Mad Catz Mad Catz R.A.T. Pro X' the program loads successfully.
After the program loads, I can plug in 'Mad Catz Mad Catz R.A.T. Pro X' back again and Gimp seems to work as expected.

**Comment by aferrero2707** [comment]
@KristijanZic
Could you please check if the latest version still has problems with USB devices?
Thanks!

**Comment by KenLidd** [comment]
The latest Oct 19th appimage version with plugins now works fine with my USB mouse.  No more start-up crashes.  Thanks

**Comment by KristijanZic** [comment]
Just tested it. Works like a charm :)

**Comment by aferrero2707** [comment]
Good to know it, and thanks for the feedback!
